src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/event/AsyncGoalEventService.kt

```diff
@@ -0,0 +1,88 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.event
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import mu.KotlinLogging
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.TelemetryService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.GoalStatus
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Step
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.StepStatus
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.GoalEventService
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Implementation of [GoalEventService] for performing additional asynchronous actions related to [Goal] events.
+ */
+@Component
+class AsyncGoalEventService(private val telemetryService: TelemetryService) : GoalEventService {
+
+  override fun goalCreated(prisonNumber: String, createdGoal: Goal) {
+    runBlocking {
+      log.debug { "Goal created event for prisoner [$prisonNumber]" }
+      launch {
+        telemetryService.trackGoalCreateEvent(createdGoal)
+      }
+    }
+  }
+
+  override fun goalUpdated(prisonNumber: String, updatedGoal: Goal, existingGoal: Goal) {
+    runBlocking {
+      log.debug { "Goal updated event for prisoner [$prisonNumber]" }
+      // val events = resolveUpdateEvents(updatedGoal, existingGoal)
+      launch {
+        telemetryService.trackGoalUpdateEvent(updatedGoal)
```

Comment:
WIP - the plan is to call a `TimelineEventService` with a list of relevant `GoalEvent`s